### PR TITLE
[Docs] Add HLRC Async Search API documentation

### PR DESCRIPTION
**Pull request by cbuescher**
Adds documentation and a corresponding test case containing typical API usage
for the Async Search API to the High Level Rest Client.

### CODE REVIEW

File: docs/java-rest/high-level/asyncsearch/submit.asciidoc [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+The following arguments can optionally be provided:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests-file}[{api}-request-arguments]
+--------------------------------------------------
+<1> The minimum time that the request should wait before
+returning a partial result (defaults to 1 second).
+<2> The expiration time of the result (defaults to 5 days).
+<3> Whether the resources should be kept after completion or failure
```
Comment (javanna) [inline docs/java-rest/high-level/asyncsearch/submit.asciidoc]: s/resources/results ? I find this description confusing, shall we add what we have in the javadocs or rest spec?

File: docs/java-rest/high-level/asyncsearch/submit.asciidoc [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+==== Optional arguments
+The following arguments can optionally be provided:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests-file}[{api}-request-arguments]
+--------------------------------------------------
+<1> The minimum time that the request should wait before
+returning a partial result (defaults to 1 second).
+<2> The expiration time of the result (defaults to 5 days).
```
Comment (javanna) [inline docs/java-rest/high-level/asyncsearch/submit.asciidoc]: it's more the validity of the request, not just of the result.

File: docs/java-rest/high-level/asyncsearch/get.asciidoc [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+==== Optional arguments
+The following arguments can optionally be provided:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests-file}[{api}-request-arguments]
+--------------------------------------------------
+<1> The minimum time that the request should wait before 
+returning a partial result (defaults to no wait).
+<2> The expiration time of the result (defaults to none).
```
Comment (javanna) [inline docs/java-rest/high-level/asyncsearch/get.asciidoc]: same comment as in the submit API

**Review by javanna — APPROVED** [review]
left a couple of small comments, LGTM otherwise

**Comment by cbuescher** [comment]
@javanna thanks for the review, I pushed changes where you asked for them. Also I found that the `keep_on_completion` default was also still set to "true" in the rest-specs so I also changes that. Take a look if you want to make sure this is correct, I can also pull this out to a different PR but I don't think its necessary.

**Comment by cbuescher** [comment]
@elasticmachine run elasticsearch-ci/default-distro

File: x-pack/plugin/src/test/resources/rest-api-spec/api/async_search.submit.json [hunk 4]
```diff
@@ -34,7 +34,7 @@
       },
       "keep_on_completion":{
         "type":"boolean",
-        "description":"Control whether the response should be stored in the cluster if it completed within the provided [wait_for_completion] time (default: true)",
+        "description":"Control whether the response should be stored in the cluster if it completed within the provided [wait_for_completion] time (default: false)",
```
Comment (javanna) [inline x-pack/plugin/src/test/resources/rest-api-spec/api/async_search.submit.json]: LOL, I changed the default one line below but not in the description, thanks for fixing

File: docs/java-rest/high-level/asyncsearch/submit.asciidoc [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+The following arguments can optionally be provided:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests-file}[{api}-request-arguments]
+--------------------------------------------------
+<1> The minimum time that the request should wait before
+returning a partial result (defaults to 1 second).
+<2> The expiration time of the request (defaults to 5 days).
+<3> Controls whether the results should be stored if request completed
```
Comment (javanna) [inline docs/java-rest/high-level/asyncsearch/submit.asciidoc]: if *the* request

**Pull request merged**